app/services/proxy_deployment_service.rb

```diff
@@ -40,10 +38,8 @@ def deploy
 
     if service_mesh_integration?
       deploy_v2 && deploy_production_v2
-    elsif apicast_configuration_driven?
+    else
       deploy_v2
-    elsif @v1_compatible
-      deploy_v1
     end
```

Comment:
I wonder if we shall raise some error in case no condition matches.

Reply:
I think the business logic should strip the `v1_compatible`
Why would you just remove the option without removing the logic, and where it is called.

Reply:
If I recall correctly, we encapsulated all deployment logic within this Service so there should not be any other usage outside of it. Removing the argument `v1_compatible` and the V1Service will delete all related logic.

Reply:
Still this is changing original logic from `if elsif elsif` to `if else`. What if we deploy neither service mesh, nor apicast v2? I heard about other integrations.

Reply:
I have reverted the change, not sure whether to raise some error if we deploy neither service mesh, nor apicast v2?
 please give your opinion @hallelujah

Reply:
I agree with @akostadinov we should add an `else` (with a test).

Reply:
So as per this line https://github.com/3scale/porta/blob/9cfb6e7dbf79dc72ce19c93e2fbc2db75b0dec28/app/services/proxy_deployment_service.rb#L39

We should only go inside this method if it is either Service Mesh, APIcast hosted or APIcast self_managed see:

https://github.com/3scale/porta/blob/9cfb6e7dbf79dc72ce19c93e2fbc2db75b0dec28/app/models/proxy.rb#L466-L468
and follow references:

https://github.com/3scale/porta/blob/9cfb6e7dbf79dc72ce19c93e2fbc2db75b0dec28/app/models/proxy.rb#L467
https://github.com/3scale/porta/blob/9cfb6e7dbf79dc72ce19c93e2fbc2db75b0dec28/app/models/service.rb#L143

So it is safe not to add another branching condition in case of other integrations.

Also we should just remove the `v1_compatible: true` because this branch as explained earlier is not needed anymore